### PR TITLE
[Tests] Block builder property test

### PR DESCRIPTION
**Pull request by dkijania**
Property tests for block builder :
- genesis block
- bft block
- genesis praos block

### CODE REVIEW

File: chain-impl-mockchain/src/block/mod.rs [hunk 1]
```diff
@@ -222,6 +222,10 @@ mod test {
         fn block_serialization_bijection(b: Block) -> TestResult {
             property::testing::serialization_bijection(b)
         }
+
+        fn block_is_consient(block: Block) -> TestResult {
```
Comment (vincenthz) [inline chain-impl-mockchain/src/block/mod.rs]: typo

Comment (vincenthz) [reply chain-impl-mockchain/src/block/mod.rs]: I couldn't find a use for this function, is there something missing maybe ?

Comment (dkijania) [reply chain-impl-mockchain/src/block/mod.rs]: Removing

File: chain-impl-mockchain/src/testing/arbitrary/utils.rs [hunk 2]
```diff
@@ -71,3 +72,39 @@ pub fn choose_random_indexes<G: Gen>(gen: &mut G, upper_bound: usize) -> HashSet
...
+    pub fn verify(&mut self, is_true: bool, desc: &str) {
+        let result = match is_true {
+            false => TestResult::error(format!("expected {} to be true, but got false", desc)),
+            true => TestResult::passed(),
+        };
+        self.0.push(result);
+    }
+
+    pub fn get_result(&self) -> TestResult {
+        match self.0.iter().cloned().find(|x| x.is_failure()) {
```
Comment (CodeSandwich) [inline chain-impl-mockchain/src/testing/arbitrary/utils.rs]: Nitpick: `self.0.iter.find(TestResult::is_failure).cloned().unwrap_or_else(TestResult::passed)`?

**Pull request merged**